CONTRIBUTING.rst

```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
+
+.. code:: bash
+
+   python -m venv <name_of_your_virtual_environment>
+
+------------------------
+Installing dependencies
+------------------------
+.. _windows-installing-dependencies:
+
+Once your virtual environment is set up, activate your virtual environment
+by executing the command: ``./venv/Scripts/Activate.ps1``. Then, install
+the dependencies for your virtual environment by running the following command:
+
+- To install development dependencies:
+
+   .. code:: bash
+
+      pip install -r .\requirements\dev-requirements.txt
+
+- To install NLP related dependencies:
+
+   .. code:: bash
+
+      pip install -r .\requirements\nlp-requirements.txt
+
+- To install NLP properties related dependencies:
+
+   .. code:: bash
+
+      pip install -r .\requirements\nlp-prop-requirements.txt
+
+- To install vision development dependencies:
+
+   .. code:: bash
+
+      pip install -r .\requirements\vision-requirements.txt
+
+..
+
+   If you are working with NLP module, then you may require to install
+   ``fasttext`` on your machine. To install, you can download a suitable
+   version of ``fasttext`` wheel file from https://www.lfd.uci.edu/~gohlke/pythonlibs/#fasttext.
+   Once downloaded, you can run the following command to install the wheel file:
+
+   .. code:: bash
+      
+      pip install <name_of_your_wheel_file>.whl
+
+Once you have installed all the dependencies, you are ready to
+work on the project.
+
+-------------------
+Running Unit Tests
+-------------------
+.. _windows-running-unit-tests:
+
+To verify and execute all the unit tests, run the following command:
+
+.. code:: bash
+
+   pylint .\tests
+
+If you want to execute specific tests, execute the commands as follows:
+
+.. code:: bash
+   
+   # Execute NLP tests 
+   pytest .\tests\nlp\
+
+   # Execute vision tests 
+   pytest .\tests\vision\
+   
+   # Execute tabular tests 
+   pytest .\tests\tabular\
+
+------------
 Coding Style
 ------------
+.. _windows-coding-style:
 
-Changes to Python code should pass both linting and docstring check, to
-make it easier with our contributors, we've included a ``makefile``
-which help you get your code style on point. in order to validate that
-your code style, you can run
+Changes to Python code should pass both linting and docstring check.
+In order to validate your code style, you can run the following commands:
 
 .. code:: bash
 
-   make validate
+   # To run pylint on all the files
+   pylint deepchecks
+
+   # To run pylint on specific file
+   pylint .\deepchecks\nlp\utils\text_properties.py
+
+   # To run pydocstyle on all the files
+   python -m pydocstyle --convention=pep257 --add-ignore=D107 deepchecks
+
+
+Linux and Mac Users
+--------------------
+For contributors using either Linux or Mac operating system, we have included
+a ``makefile`` which help you get all the setup done in just a few commands.
+The ``make`` commands will create the virtual environment and install all the
```

Comment:
What will just running "make" do? Did you test that?

Reply:
No what I meant here was that all the make commands will have built-in commands to create the virtual env and no manual creation is required.